File: cpp/problems/leetcode/1207-Unique-Number-of-Occurences.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
 public:
  bool uniqueOccurrences(vector<int>& arr) {
    unordered_map<int, int> count;

    for (int val : arr) {
      if (count.find(val) != count.end()) {
        count[val]++;
      } else {
        count[val] = 1;
      }
    }
    unordered_set<int> set_count;
    for (auto [key, value] : count) {
      set_count.insert(value);
    }

    cout << count.size() << " " << set_count.size() << "\n";
    return count.size() == set_count.size();
  }
};

File: python/problems/leetcode/1207-Unique-Number-of-Occurences.py
from typing import List


class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        pair = {}

        for i in arr:
            if i in pair:
                pair[i] += 1
            else:
                pair[i] = 1

        return len(set(pair.values())) == len(pair)
